packages/core/useFetch/index.ts

```diff
@@ -87,6 +87,8 @@ export interface UseFetchOptions {
    * @default false
    */
   refetch?: MaybeRef<boolean>
+
+  beforeFetch?: (url: string, options: RequestInit) => Promise<{ url: string; options: RequestInit }>
```

Comment:
Think we can have non-promise returns as well. Do you have any plan to support canceling the fetch in this callback?

Reply:
ah yeah, we should be able to do non-promise returns, I'll add that. As for canceling we can probably pass a third parameter to abort the request

Reply:
I am thinking if we could unify the arguments and returns signature. Something like

```ts
beforeFetch?: (ctx: Context) => Promise<Context> | Context
```

Reply:
Would you include the abort option in that context? Seems like it would make sense to have something like 

```ts
type Context = { url: string; options: RequestInit }
beforeFetch?: (ctx: Context, abort: Fn) => Promise<Context> | Context
```

Reply:
How about 

```ts
beforeFetch?: (ctx: Context) => Promise<Partial<Context>> | Partial<Context>
```

I am not sure about either way tbh 😂

Reply:
Hmm yeah partial should be fine I think